### PR TITLE
HOSS2-321 | form API 

### PR DESCRIPTION
**Pull request by liampan**

### CODE REVIEW

File: test/services/SessionCacheServiceSpec.scala [hunk 1]
```diff
@@ -47,8 +46,8 @@ class SessionCacheServiceSpec extends PlaySpec with BeforeAndAfterEach with Scal
 
   val formModel = NinoSearchFormModel(
     nino = NinoGenerator.generateNino,
-    givenName = "Pan",
-    familyName = "Oldfield",
+    givenName = "Peter",
+    familyName = "Pan",
```
Comment (burns238) [inline test/services/SessionCacheServiceSpec.scala:50]: Is this change necessary?

File: app/forms/SearchByMRZForm.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    )(MrzSearchFormModel.apply)(MrzSearchFormModel.unapply)
+  }
+}
+
+object SearchByMRZForm {
+  final val AllowedDocumentTypes = Seq("PASSPORT", "NAT", "BRC", "BRP")
+
+  final val DocumentNumberMaxLength = 30
+
+  val CountryList = Seq("TODO WHERE ARE WE GETTING THIS FROM", "AFG") //todo iso-3166 list
```
Comment (liampan) [inline app/forms/SearchByMRZForm.scala:51]: Need to work out where our list of iso-3166 countries are coming from, however i dont think affects this PR. have added AFG so we can test until its implimented.

File: test/services/SessionCacheServiceSpec.scala (hunk 1 above)
Comment (liampan) [reply test/services/SessionCacheServiceSpec.scala:50]: Any better?
```suggestion
    givenName = "mike",
    familyName = "Burns",
```

File: app/forms/SearchByMRZForm.scala [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  def apply(): Form[MrzSearchFormModel] = Form[MrzSearchFormModel] {
+    mapping(
+      "documentType" -> nonEmptyText("documentType")
+        .transform[String](_.toUpperCase, identity)
+        .verifying("error.documentType.invalid", AllowedDocumentTypes.contains(_)),
+      "documentNumber" -> nonEmptyText("documentNumber")
+        .verifying(
+          "error.documentNumber.invalid",
+          dn => dn.length <= DocumentNumberMaxLength && dn.forall(c => c.isDigit || c.isLetter)),
```
[the remark this answers is not in the document]
Comment (liampan) [reply app/forms/SearchByMRZForm.scala]: Have added hyphen, good spot. 
I tend to avoid regex if at all possible as I know some devs fear it, but happy to change if youd prefer?

**Review by burns238 — APPROVED** [review]
